### PR TITLE
Correct Meraki Interface Descriptions

### LINKED ISSUE #1306: Meraki SSoT - when syncing interfaces SVI from Meraki devices, nautobot interface `description` field is empty
### Environment
* Python version:   3.13
* Nautobot version:   3.1.x
* nautobot-ssot version:   4.5.2

### Expected Behavior

At the end of the Meraki SSoT sync, interface `description` field corresponding to svi interface is blank. This makes it difficult to understand what's the real purpose of the svi.

It would likely make more sense to use the `name` field instead as the port `description`, since vlan has no real `description` field, cf. the [getNetworkApplianceVlans](https://developer.cisco.com/meraki/api-v1/get-network-appliance-vlans/) api which is used in `load_lan_svis()` called by `load_firewall_ports()` in `nautobot_ssot/integrations/meraki/diffsync/adapters/meraki.py`.

There is no `description` coming back from `getNetworkApplianceVlans` and the vlan `name` is often enough to understand its real purpose.

### Steps to Reproduce
1. perform a `Meraki to Nautobot` SSoT sync
2. any svi existing on a `MX/MG/Z` device will be synced in nautobot with a blank `description` 

cf. NTC-6320

### PR DESCRIPTION
**Pull request by zackt25**
Closes: [#1306](https://github.com/nautobot/nautobot-app-ssot/issues/1306)

## What's Changed

This pull request implements robust synchronization of interface descriptions for Meraki devices, ensuring that interface descriptions in Nautobot are consistently sourced from the most appropriate Meraki API fields and accurately reflect the state of the Meraki environment. It fixes issues with blank descriptions, ensures descriptions are tracked and updated, and enhances test coverage and documentation to reflect the new behavior.

**Interface Description Synchronization Improvements:**

* Interface descriptions for Meraki MS switchports are now sourced from the Meraki port `name`, and SVI descriptions for MX/MG/Z devices are sourced from the VLAN `name`. Descriptions for interfaces without a corresponding Meraki label (e.g., WAN uplinks, management ports) are set to empty strings, with clear documentation and warnings added. [[1]](diffhunk://#diff-20e76ce6e29af2b206aea08424b2b0f3818d963a90a834173bbe8e5cec62c6c8R1) [[2]](diffhunk://#diff-09dd61a5a28213caaa6c28c568375eeaf1636cd4072d4451d87d78e6cce5443cR1-R2) [[3]](diffhunk://#diff-2a051c8366cd1690cdfe9d893e3fa5c3970d4050bcf466620346334e97cedb6cR83-R96)

* The `Port` DiffSync model now tracks the `description` attribute, ensuring that interface descriptions are no longer silently discarded during sync. [[1]](diffhunk://#diff-09dd61a5a28213caaa6c28c568375eeaf1636cd4072d4451d87d78e6cce5443cR1-R2) [[2]](diffhunk://#diff-804bf71af90679d04edc2658fd6313e25d60b985dedd57b749435d7549032e74L87-R92)

**Bug Fixes:**

* Fixed blank interface descriptions for SVIs by sourcing from the VLAN name, and ensured that switchport descriptions are correctly populated from the Meraki port name or set to empty if missing. [[1]](diffhunk://#diff-09dd61a5a28213caaa6c28c568375eeaf1636cd4072d4451d87d78e6cce5443cR1-R2) [[2]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570L788-R802) [[3]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R612) [[4]](diffhunk://#diff-6d80202c44819b7692d4ca8ba6eacda67a61f9a547a2b764567f6981393fdb15R427-R474)

* Ensured that interface descriptions are correctly propagated and updated in Nautobot, both on creation and update, and that they default to empty strings if missing. [[1]](diffhunk://#diff-0e08b892adb655cb71fd597c9d2cf73066e0e008375c9c5c8524a1f19ad81946R232) [[2]](diffhunk://#diff-0e08b892adb655cb71fd597c9d2cf73066e0e008375c9c5c8524a1f19ad81946R248-R249) [[3]](diffhunk://#diff-c98d84805028b662cd15255aa40fdfce78e30e7233bf474bece48204b41faf22R170)

**Test and Fixture Updates:**

* Added and updated tests to verify that interface descriptions are correctly set from Meraki data, including edge cases where names are missing or null. Updated fixtures to provide realistic port names. [[1]](diffhunk://#diff-6d80202c44819b7692d4ca8ba6eacda67a61f9a547a2b764567f6981393fdb15R370-R372) [[2]](diffhunk://#diff-6d80202c44819b7692d4ca8ba6eacda67a61f9a547a2b764567f6981393fdb15R427-R474) [[3]](diffhunk://#diff-6d80202c44819b7692d4ca8ba6eacda67a61f9a547a2b764567f6981393fdb15R412-R414) [[4]](diffhunk://#diff-fb959b8c5e538f85bc4a9db3bcce35de8a0b4b8b6fa7052d4357ba2095b92369R93) [[5]](diffhunk://#diff-fb959b8c5e538f85bc4a9db3bcce35de8a0b4b8b6fa7052d4357ba2095b92369R138-R140)

**Documentation:**

* Expanded the Meraki integration documentation to clearly explain how interface descriptions are mapped, including caveats and authoritative behavior warnings.

**Code Consistency:**

* Refactored all relevant adapter code to consistently set and pass the `description` attribute for all interface types, defaulting to empty strings where appropriate. [[1]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R213) [[2]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R265) [[3]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R310) [[4]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R335) [[5]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R379) [[6]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R419) [[7]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R459) [[8]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R499) [[9]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R539) [[10]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R579) [[11]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R612) [[12]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R634) [[13]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570R685) [[14]](diffhunk://#diff-3aca2721d4740031acc53f775d9232185cc6df9433082e15b72d35ca4851d570L820-L832)

These changes ensure that interface descriptions in Nautobot accurately reflect Meraki configuration and improve the reliability and clarity of Meraki SSoT integration.

### CODE REVIEW

File: nautobot_ssot/integrations/meraki/diffsync/models/nautobot.py [hunk 1]
```diff
@@ -244,6 +245,8 @@ def create(cls, adapter, ids, attrs):
     def update(self, attrs):
         """Update Interface in Nautobot from NautobotDevice object."""
         port = Interface.objects.get(id=self.uuid)
+        if "description" in attrs:
+            port.description = attrs["description"] or ""
```
Comment (jdrew82) [inline nautobot_ssot/integrations/meraki/diffsync/models/nautobot.py]: The or default isn't necessary here as you've already confirmed the key is in the attrs dict.

**Review by jdrew82 — CHANGES_REQUESTED** [review]
Just a small tweak.

File: nautobot_ssot/integrations/meraki/diffsync/models/nautobot.py (hunk 1 above)
Comment (zackt25) [reply nautobot_ssot/integrations/meraki/diffsync/models/nautobot.py]: Fixed with https://github.com/nautobot/nautobot-app-ssot/pull/1319/commits/6822759c4efef7ffbfd687c8e08563bf6dbd2bcc

Comment (zackt25) [reply nautobot_ssot/integrations/meraki/diffsync/models/nautobot.py]: Would meraki ever send `{"description": None}` ?

Comment (zackt25) [reply nautobot_ssot/integrations/meraki/diffsync/models/nautobot.py]: Here is the issue and why default was needed.

In the DiffSync model, None is legal:

description: Optional[str] = None     # base.py:92 — pydantic accepts None

In the Django column, it isn't:

description = models.CharField(max_length=CHARFIELD_MAX_LENGTH, blank=True)
